Lib/test/pickletester.py

```diff
@@ -3081,6 +3094,19 @@ def reduce_2(obj):
         self.assertEqual(default_load_dump(a), REDUCE_A)
         self.assertIsInstance(default_load_dump(b), BBB)
 
+        # End-to-end testing of save_reduce with the state_setter keyword
+        # argument. This is a dispatch_table test as state_setter is useful to
+        # register custom state-setting behavior for classes that were not
+        # created by the user.
```

Comment:
"for classes that were not created by the user."

=>

"when the default behavior of the `__setitem__` method is not flexible enough."